File: AggieAutoSchedulerAngular/ClientApp/src/app/classes/course.ts
import { Period } from './period';
import { Exam } from './exam';
import { CdkRowDef } from '@angular/cdk/table';

export class Course {
  crn: number;
  subject: string;
  courseNumber: number;
  sectionNumber: number;
  credits: number;
  title: string;
  monday: Period[] = [];
  tuesday: Period[] = [];
  wednesday: Period[] = [];
  thursday: Period[] = [];
  friday: Period[] = [];
  exams: Exam[] = [];

  constructor(
    crn: number, subject: string, courseNumber: number,
    sectionNumber: number, credits: number, title: string,
    monday: Period[] = [], tuesday: Period[] = [], wednesday: Period[] = [],
    thursday: Period[] = [], friday: Period[] = [], exams: Exam[] = []
  ) {
    this.crn = crn;
    this.subject = subject;
    this.courseNumber = courseNumber;
    this.sectionNumber = sectionNumber;
    this.credits = credits;
    this.title = title;
    this.monday = monday;
    this.tuesday = tuesday;
    this.wednesday = wednesday;
    this.thursday = thursday;
    this.friday = friday;
    this.exams = exams;
  }


}

File: AAS.DB/Course.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace AAS.DB
{
    public class Course
    {
        public Course()
        {
            Periods = new List<Period>();
            Exams = new List<Exam>();
        }

        [Key]
        public int ID { get; set; }
        public int CRN { get; set; }
        public string Subject { get; set; }
        public int Credits { get; set; }
        public string Title { get; set; }
        public int CourseNumber { get; set; }
        public int SectionNumber { get; set; }
        public List<Period> Periods { get; set; }
        public List<Exam> Exams { get; set; }

    }
}
